internal/artifacts/schematic.go

```diff
@@ -103,11 +108,24 @@ func schematicExtension(schematicID string) (io.Reader, error) {
 	if err = tw.WriteHeader(&tar.Header{
 		Name:     filepath.Join("rootfs/usr/local/share/schematic", schematicID), // empty file
 		Typeflag: tar.TypeReg,
-		Mode:     0o755,
+		Mode:     0o644,
 	}); err != nil {
 		return nil, fmt.Errorf("failed to write rootfs header: %w", err)
 	}
 
+	if err = tw.WriteHeader(&tar.Header{
+		Name:     "rootfs/usr/local/share/schematic/schematic.yaml",
```

Comment:
this would be hard to fetch back from Omni. we can keep it this way as well, but I think we should rather embed the schematic into some field of the `extensions.Metadata` above, so that we can get a schematic back by doing `talosctl get extensions`

Reply:
I was thinking omni could read back the file?

Reply:
should we change the signature of `extensions.Metadata`?

Reply:
Omni can't read files in maintenance mode

Reply:
hmm maybe i can add another virtual extensions with the name `overlay` and set the `version` field as `<image>:<name>` or something like that

Reply:
hmm, that doesn't work since, we need `options` too

Reply:
> Omni can't read files in maintenance mode

didn't we land support for 1.7 to allow over maintenance? and since overlays are for 1.7+ this should work I believe

Reply:
no, this doesn't work. I wonder if we could re-use the description or any other free-form field (or add another one) just to put there the schematic as string as YAML